tests/test_views.py

```diff
@@ -4583,5 +4529,269 @@ def test_confirm_user_signal_called_when_user_confirms_email(self):
+            )
+            d.save()
+
+        found = [self.draft]
+        # Drafts for self.node
+        for i in range(3):
+            d = DraftRegistrationFactory(
+                initiator=self.user,
+                branched_from=self.node,
+                meta_schema=self.meta_schema,
+                schema_data={}
+            )
+            d.save()
+            found.append(d)
+        url = self.node.api_url_for('get_draft_registrations')
+
+        res = self.app.get(url, auth=self.auth)
+        assert_equal(res.status_code, http.OK)
+        # 3 new, 1 from setUp
+        assert_equal(len(res.json['drafts']), 4)
+        for draft in res.json['drafts']:
+            assert_in(draft['pk'], [f._id for f in found])
+
+    def test_create_draft_registration(self):
+        target = NodeFactory(creator=self.user)
+        metadata = {
+            'summary': {'value': 'Some airy'}
+        }
+        payload = {
+            'schema_name': 'Open-Ended Registration',
+            'schema_version': 1,
+            'schema_data': metadata,
+        }
+        url = target.api_url_for('create_draft_registration')
+
+        res = self.app.post_json(url, payload, auth=self.auth)
+        assert_equal(res.status_code, http.CREATED)
+        draft = DraftRegistration.find_one(Q('branched_from', 'eq', target))
+        assert_equal(draft.registration_schema, self.meta_schema)
+        assert_equal(draft.registration_metadata, metadata)
+
+    def test_new_draft_registration(self):
+        target = NodeFactory(creator=self.user)
+        payload = {
+            'schema_name': 'Open-Ended Registration',
+            'schema_version': 1
+        }
+        url = target.web_url_for('new_draft_registration')
+
+        res = self.app.post(url, payload, auth=self.auth)
+        assert_equal(res.status_code, http.FOUND)
+        draft = DraftRegistration.find_one(Q('branched_from', 'eq', target))
+        assert_equal(draft.registration_schema, self.meta_schema)
+
+    def test_update_draft_registration(self):
+        metadata = {
+            'summary': {'value': 'updated'}
+        }
+        assert_not_equal(metadata, self.draft.registration_metadata)
+        payload = {
+            'schema_data': metadata,
+            'schema_name': 'OSF-Standard Pre-Data Collection Registration',
+            'schema_version': 1
+        }
+        url = self.node.api_url_for('update_draft_registration', draft_id=self.draft._id)
+
+        res = self.app.put_json(url, payload, auth=self.auth)
+        assert_equal(res.status_code, http.OK)
+
+        open_ended_schema = MetaSchema.find_one(
+            Q('name', 'eq', 'OSF-Standard Pre-Data Collection Registration') &
+            Q('schema_version', 'eq', 1)
+        )
+        self.draft.reload()
+        assert_equal(open_ended_schema, self.draft.registration_schema)
+        assert_equal(metadata, self.draft.registration_metadata)
+
+    def test_delete_draft_registration(self):
```

Comment:
It may be worth adding some tests of permissions- eg *only_admin_can_delete_drafts* or such.